### PR TITLE
Use blocking thread pool in the `JettyClient#resource`

### PR DESCRIPTION
**Pull request by danicheg**
`org.eclipse.jetty.util.component.AbstractLifeCycle#stop` uses under the hood synchronized lock, so blocking thread pool is more desired to use.

### CODE REVIEW

**Review by bplommer — APPROVED** [review]

File: jetty-client/src/main/scala/org/http4s/jetty/client/JettyClient.scala [hunk 1]
```diff
@@ -62,9 +62,10 @@ object JettyClient {
...
         }
       )
-    val dispose = F
-      .delay(client.stop())
-      .handleErrorWith(t => F.delay(logger.error(t)("Unable to shut down Jetty client")))
+    val dispose =
+      F
+        .blocking(client.stop())
+        .handleErrorWith(t => F.delay(logger.error(t)("Unable to shut down Jetty client")))
     Resource.make(acquire)(_ => dispose)
```
Comment (bplommer) [inline jetty-client/src/main/scala/org/http4s/jetty/client/JettyClient.scala:69]: Kinda unrelated but shouldn't we dispose regardless of whether acquire succeeds?
```suggestion
    Resource.eval(acquire).onFinalize(dispose)
```

Comment (danicheg) [reply jetty-client/src/main/scala/org/http4s/jetty/client/JettyClient.scala:69]: Probably that would mean behavior change. According to the contract, `client` may come to the `resource` from outside.

Comment (rossabaker) [reply jetty-client/src/main/scala/org/http4s/jetty/client/JettyClient.scala:69]: I guess we want to stop it if and only if we start it?

Comment (bplommer) [reply jetty-client/src/main/scala/org/http4s/jetty/client/JettyClient.scala:69]: Yeah, that makes sense.

**Pull request merged**